### PR TITLE
retail edge demo (red)

### PR DESCRIPTION
**Pull request by bartr**
## Purpose of PR

- added scripts to create / manage our demo clusters (red-vms)
- fixed linter issues
- fixed ssl setup
  - added the ca certs to the pem file
  - added basic instructions with a todo tag
- moved tls secret creation before flux
- added akdc check ssl
  - this also checks the DNS
  - this hits https://$(hostname).cseretail.com/tinybench/17 so it's a complete test
  - tinybench is set to deploy to all clusters (it's our "heartbeat")
- added cert to Codespaces secret
  - they get written to ~/.ssh/certs.*
- added shared SSH key to Codespaces secret
  - gets written to ~/.ssh/id_rsa.*
- added bash linter to Codespaces
- added long timeouts to first ssh and scp commands
  - I haven't seen any scp timeouts since
- updated gen-gitops
  - support the red servers
  - generate the expandTargets.json file for AutoGitOps

# Type of PR

- [x] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/retail-edge/issues/58
- Closes https://github.com/retaildevcrews/retail-edge/issues/64
- Closes https://github.com/retaildevcrews/retail-edge/issues/65
- Closes https://github.com/retaildevcrews/retail-edge/issues/68
- Closes https://github.com/retaildevcrews/retail-edge/issues/71
- Closes https://github.com/retaildevcrews/retail-edge/issues/74
- Closes https://github.com/retaildevcrews/retail-edge/issues/75
- Closes https://github.com/retaildevcrews/retail-edge/issues/76

### CODE REVIEW

File: src/cli/create-cluster [hunk 1]
```diff
@@ -27,21 +32,23 @@ declare ShowHelp=false
...
 # register arguments
 shArgs.arg "DnsZone" -z --zone PARAMETER true
-shArgs.arg "CertCRTPath" -c --crt PARAMETER true
+shArgs.arg "CertCRTPath" -c --cert PARAMETER true
 shArgs.arg "CertKeyPath" -k --key PARAMETER true
 shArgs.arg "Location" -l --location PARAMETER true
-shArgs.arg "UseSSL" -s --ssl FLAG true 
-shArgs.arg "ShowHelp" -h --usage FLAG true 
+shArgs.arg "UseSSL" -s --ssl FLAG true
+shArgs.arg "ShowHelp" -h --help FLAG true
```
Comment (hattan) [inline src/cli/create-cluster:42]: This was intentional due to an issue with --help . This was a bug in shArgs. I just fixed the bug and pushed an update to address it so --help should work.

**Review by hattan — APPROVED** [review]
LGTM

File: src/cli/create-cluster [hunk 2]
```diff
@@ -196,39 +202,36 @@ usage() {
...
-  cat << "EOF"
-
- █████  ██   ██ ██████   ██████ 
-██   ██ ██  ██  ██   ██ ██      
-███████ █████   ██   ██ ██      
-██   ██ ██  ██  ██   ██ ██      
-██   ██ ██   ██ ██████   ██████ 
-
-EOF
+  echo " █████  ██   ██ ██████   ██████"
```
[the remark this answers is not in the document]
Comment (bartr) [reply src/cli/create-cluster:222]: the code folding wasn't working correctly in VS Code - I like CAT better too

We can change it back in another PR if we want - maybe put it in a separate file?

[the remark this answers is not in the document]
Comment (hattan) [reply src/cli/create-cluster:222]: That's a good idea. I like keeping it in a separate file. I'll create a small PR to patch.

**Pull request merged**